### PR TITLE
Remove hardcoded note and fix reporting period-specific messages

### PR DESCRIPTION
**Pull request by ccostino**
Finalizes #672 

This changeset finalizes the new timecard notes feature by removing the hardcoded message at the top of the timecard form and making it an initial model available for immediate use.  It also fixes the display of the reporting-period specific messages and provides a few extra options for them as well:

* An optional title
* A style to choose from (same as timecard notes)
* Whether or not the message is actually enabled for display in someone's timecard

These messages will appear above any general messages as they are reporting period specific if they are enabled.  If they are not, they can still be useful for an admin for internal notes.

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -3,3 +3,5 @@ django-debug-toolbar==1.8
 django-webtest==1.7.9
 factory-boy==2.8.1
 nplusone==0.8.1
+pdbpp==0.9.1
+ipython==6.2.1
```
Comment (annalee) [inline requirements-dev.txt]: Do we need these in dev requirements?

Comment (ccostino) [reply requirements-dev.txt]: Nope!  I find them incredibly useful for helping with debugging and development, but if you rather I keep it local, happy to take them back out.

Comment (annalee) [reply requirements-dev.txt]: I find them useful too, but probably best to keep them local so as not to annoy folks who have a different system. Plus fewer dependencies = easier ATO.

Comment (ccostino) [reply requirements-dev.txt]: Out they go!

**Pull request merged**